location/src/main/java/org/motechproject/nms/location/service/impl/LocationServiceImpl.java

```diff
@@ -9,4 +11,20 @@
 @Service("locationService")
 public class LocationServiceImpl implements LocationService {
 
+    public static final int HACKY_LANGUAGE_CODE = 34;
+    public static final long HACKY_WHITELISTED_STATE_CODE = -1L;
+    public static final long HACKY_NON_WHITELISTED_STATE_CODE = -10L;
+
+    @Override
+    public State getStateForLanguage(Language language) {
+        // TODO: #71 For a given languageLocationCode find the districts it maps to and find the state for
```

Comment:
language location code maps to a set of districts, or to a state.

NMS.GEN.LANG.001

If there is only one “Language-location code” in the state, all districts
shall have same “Language-location code”; else, multiple groups shall be
formed, each group having a set of districts and having a unique
“Language-location code”.
Division or reorganization of geographical location for “Language-location
code” mapping is not supported and an item for future.

On Thu, Apr 23, 2015 at 11:48 AM, kosh <notifications@github.com> wrote:

> In
> location/src/main/java/org/motechproject/nms/location/service/impl/LocationServiceImpl.java
> <https://github.com/koshalt/mim/pull/74#discussion_r28991554>:
>
> > @@ -9,4 +11,20 @@
> >  @Service("locationService")
> >  public class LocationServiceImpl implements LocationService {
> >
> > +    public static final int HACKY_LANGUAGE_CODE = 34;
> > +    public static final long HACKY_WHITELISTED_STATE_CODE = -1L;
> > +    public static final long HACKY_NON_WHITELISTED_STATE_CODE = -10L;
> > +
> > +    @Override
> > +    public State getStateForLanguage(Language language) {
> > +        // TODO: #71 For a given languageLocationCode find the districts it maps to and find the state for
>
> I thought that the whole point of language location code was that it would
> only map to one unique district? (vs. a language code that could map to
> districts across multiple states?)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/koshalt/mim/pull/74/files#r28991554>.
>